src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -1849,6 +1849,34 @@ void c_typecheck_baset::typecheck_expr_side_effect(side_effect_exprt &expr)
   }
 }
 
+/// type2name generates strings that aren't valid C identifiers
+/// If we want utilities like dump_c to work properly identifiers
+/// should ideally always be valid C identifiers
+/// This replaces some invalid characters that can appear in type2name output.
+static std::string clean_type_name(const std::string &type_name)
```

Comment:
:grey_question: Could (should?) this be fixed in type_to_name? Seems odd that the parsing would be concerned about valid C names. 

I think the answer is no: type_to_name is making valid C stuff like `int*`, and then this code is misusing that to put straight back into a C identifier.

Reply:
We didn't want to touch `type_to_name`, especially for this fix, because we don't know how much depends on the output it produces, and we were afraid of breaking a lot more than we could fix.